llama.cpp

```diff
@@ -689,6 +695,17 @@ struct llama_model_loader {
         }
         ggml_set_name(tensor, lt.name.c_str());
         LLAMA_ASSERT(lt.ggml_tensor == NULL); // if this fails, we called get_tensor twice on the same tensor
+
+#ifdef GGML_USE_CUBLAS
+        if (backend == GGML_BACKEND_GPU || backend == GGML_BACKEND_GPU_SPLIT) {
+            struct ggml_tensor_extra_gpu * extra = new struct ggml_tensor_extra_gpu;
```

Comment:
Have to deallocate these extra buffers at the end of the program

Reply:
They are deallocated [here](https://github.com/JohannesGaessler/llama.cpp/blob/11af67866e927ddab9dd86735a7e140fcc93a253/llama.cpp#L203-L207) and [here](https://github.com/JohannesGaessler/llama.cpp/blob/11af67866e927ddab9dd86735a7e140fcc93a253/ggml-cuda.cu#L1141-L1158) when `llama_model` gets deconstructed